CoreFoundation/Base.subproj/CFInternal.h

```diff
@@ -229,7 +229,7 @@ extern void __CFGenericValidateType_(CFTypeRef cf, CFTypeID type, const char *fu
 #define __CFBitfield64GetValue(V, N1, N2)	(((V) & __CFBitfield64Mask(N1, N2)) >> (N2))
 #define __CFBitfield64SetValue(V, N1, N2, X)	((V) = ((V) & ~__CFBitfield64Mask(N1, N2)) | ((((uint64_t)X) << (N2)) & __CFBitfield64Mask(N1, N2)))
 
-#if __LP64__ || DEPLOYMENT_TARGET_ANDROID
+#if TARGET_RT_64_BIT || DEPLOYMENT_TARGET_ANDROID
```

Comment:
TARGET_ are usually only from TargetConditionals.h which CoreFoundation does not control on Darwin. So we probably need to first define something along the lines of CFRUNTIME_64BIT. Additionally this should be defined if it does not get defined in a public header; CFBase.h is a good spot to do this.

```c
#ifndef CFRUNTIME_64BIT
#if __LLP64__ || __LP64__
#define CFRUNTIME_64BIT 1
#else
#define CFRUNTIME_64BIT 0
#endif
#endif
```

Reply:
hmm I take that back: TARGET_RT_64_BIT is defined by TargetConditionals on Darwin.

Do we have that defined in <CoreFoundation/TargetConditionals.h>?

Reply:
Oh, so we shouldn't be using `TARGET_RT_64_BIT` and define and use `CF_RUNTIME_64_BIT` through out?

Reply:
Yes, that is defined in the `<CoreFoundation/TargetConditionals.h>` header.